src/java.base/share/classes/java/security/MessageDigest.java

```diff
@@ -460,7 +460,8 @@ public String toString() {
      * All bytes in {@code digesta} are examined to determine equality.
      * The calculation time depends only on the length of {@code digesta}.
      * It does not depend on the length of {@code digestb} or the contents
-     * of {@code digesta} and {@code digestb}.
+     * of {@code digesta} and {@code digestb}, unless {@code digestb} is null
+     * or has a length of zero bytes.
```

Comment:
I think this new text should be moved earlier to the first sentence, since that first sentence is not always true given this condition. I suggest rewording the first two sentences as:

"All bytes in {@code digesta} are examined to determine equality, unless {@code digestb} is {@code null}
or has a length of zero bytes. If {@code digestb} is not {@code null} and does not have a length of zero bytes,
then the calculation time depends only on the length of {@code digesta}."